k8s/controllers/k8s/globalobject_controller.go

```diff
@@ -17,29 +17,304 @@ package k8s
 
 import (
 	"context"
+	"fmt"
+	"strings"
+	"time"
 
-	"github.com/go-logr/logr"
+	"github.com/apache/skywalking-swck/operator/pkg/kubernetes"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/retry"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	v1alpha1 "github.com/v6d-io/v6d/k8s/apis/k8s/v1alpha1"
+	"github.com/v6d-io/v6d/k8s/schedulers"
+)
+
+const (
+	// MaxDuration replesents the max duration of waiting the distributed assembly globalobject
+	MaxDuration = 2 * time.Minute
 )
 
 // GlobalObjectReconciler reconciles a GlobalObject object
 type GlobalObjectReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	Scheme   *runtime.Scheme
+	Template kubernetes.Repo
+	Recorder record.EventRecorder
+}
+
+// DistributedAssemblyConfig is the config for the distributed assembly job
+type DistributedAssemblyConfig struct {
+	Name                 string
+	Namespace            string
+	StreamID             string
+	JobName              string
+	GlobalobjectID       string
+	OldObjectToNewObject string
+	VineyardSockPath     string
+}
```

Comment:
Why `DistributedAssemblyConfig`  are put in `globalobject_controller`? Looks quite strange..

The assembly, checkpoint, repartition, etc. should be a layer that implemented upon "global object".